### PR TITLE
feat: move `RethRpcConfig` to rpc-builder

### LINKED ISSUE #8460: Extract core rpc server types to standalone crate
### Describe the feature

currently node-core pulls in rpc-builder for constants and setup code

rpc-builder is pretty beefy because it pulls in the entire stack required for all endpoints.
node-core sits pretty deep in the stack so this results in bloated dependency trees

## TODO
[x] add new crate `rpc-server(-common|types)` and extract all types/constants used by node-core rn
[ ] move eth cache types https://github.com/paradigmxyz/reth/blob/13fcb885c60bab1390c4525cb5a0c35ab215e833/crates/rpc/rpc/src/eth/cache/config.rs#L2-L2

### Additional context

_No response_

### LINKED ISSUE #8602: Investigate moving core rpc server types to standalone crate
> I think next step is investigating, whether we can extract the node-core/cli/config/ traits.
> 
> a few solutions are possible here
> a) move them to node-builder
> b) move all of them to a separate crate under CLI
> c) move RethRpcConfig to rpc-builder
> 
> perhaps option c) is the most viable solution

_Originally posted by @mattsse in https://github.com/paradigmxyz/reth/pull/8597#pullrequestreview-2097204532_

### PR DESCRIPTION
**Pull request by leruaa**
Closes #8602
Closes #8460

### CODE REVIEW

File: crates/node-core/src/args/rpc_server.rs [hunk 1]
```diff
@@ -470,11 +330,11 @@ impl TypedValueParser for RpcModuleSelectionValueParser {
     }
 }
 
+#[cfg(feature = "nope")]
```
Comment (leruaa) [inline crates/node-core/src/args/rpc_server.rs:333]: TODO: remove this and fix errors

**Comment by leruaa** [comment]
@mattsse I would appreciate an early review to see if I'm in the right direction

File: crates/node-core/src/cli/config.rs [hunk 2]
```diff
@@ -1,80 +1,9 @@
...
-    RpcServerConfig, ServerBuilder, TransportRpcModuleConfig,
-};
 use reth_transaction_pool::PoolConfig;
-use std::{borrow::Cow, path::PathBuf, time::Duration};
-
-/// A trait that provides a configured RPC server.
-///
-/// This provides all basic config values for the RPC server and is implemented by the
-/// [`RpcServerArgs`](crate::args::RpcServerArgs) type.
-pub trait RethRpcConfig {
```
Comment (mattsse) [inline crates/node-core/src/cli/config.rs:18]: we can move this to rpc-builder/config and rename `RethRpcServerConfig`

File: crates/node-core/src/args/rpc_server.rs [hunk 3]
```diff
@@ -469,196 +323,3 @@ impl TypedValueParser for RpcModuleSelectionValueParser {
         Some(Box::new(values))
     }
 }
-
-#[cfg(test)]
-mod tests {
```
Comment (mattsse) [inline crates/node-core/src/args/rpc_server.rs:474]: I want to keep these tests but no longer make use of the trait functions

**Review by mattsse — CHANGES_REQUESTED** [review]
nice, getting rid of the builder dep is amazing

File: crates/node-core/src/args/rpc_server.rs (hunk 3 above)
Comment (leruaa) [reply crates/node-core/src/args/rpc_server.rs:474]: For instance replacing:
```rust
let config = args.transport_rpc_module_config();
```
by something like:
```rust
let config = TransportRpcModuleConfig::default();
```
implies moving `TransportRpcModuleConfig` and many others config types to rpc-server-types. Is that what we want?

Comment (mattsse) [reply crates/node-core/src/args/rpc_server.rs:474]: this test can be moved to where we impl serverconfig trait, so I guess we need to move some tests and keep some

**Review by mattsse — APPROVED** [review]
nice, tysm!

this helps a lot

**Pull request merged**